mlc-llm/tutorial_compile_llama2_with_mlc_llm.ipynb

```diff
@@ -0,0 +1,1029 @@
+      "source": [
+        "We use a dataclass `BuildArgs` to organize the arguments for building the model. Besides specifying the model with `model` (when the weight is local), you could also use the argument `hf_path`, which will download the model from huggingface directly.\n",
+        "\n",
+        "For more details on the arguments, please see [the docs for the CLI](https://mlc.ai/mlc-llm/docs/compilation/compile_models.html#compile-command-specification) for now. We will update documentation for `BuildArgs` soon. (Or you could look at the source code)"
+      ],
+      "metadata": {
+        "id": "PpZGTy6QgZCB"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "build_args = mlc_llm.BuildArgs(\n",
+        "    model=\"Llama-2-7b-chat-hf\",\n",
+        "    quantization=\"q4f16_1\",\n",
+        "    target=\"cuda\")"
+      ],
+      "metadata": {
+        "id": "ei3r2kenNqbF"
+      },
+      "execution_count": 7,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "`mlc_llm.build_model` is the main entrance here. It takes in a `BuildArgs` to start the entire model compilation workflow.\n",
+        "\n",
+        "**Google Colab** If you are using Colab, the line below may require more RAM than the default Colab provides. You may need to either upgrade to a paid Colab plan, or run it in other environments. (Or sometimes, when you keep rerunning, (just the build portion), it eventually builds without exceeding the RAM Colab provides)"
+      ],
+      "metadata": {
+        "id": "OIX3oyPhnCQs"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "The output of `lib_path, model_path, chat_config_path = mlc_llm.build_model(build_args)` is given as a tuple of three paths.\n",
+        "\n",
+        "`lib_path` is the path to the specific binary that has been built.\n",
+        "\n",
+        "`model_path` is the path to the folder containing the compiled model parameters and other model specific configuration needed for other `mlc` modules.\n",
+        "\n",
+        "`chat_config_path` is the path to the specific `.json` configuration needed to have this model work with `mlc_chat`."
+      ],
+      "metadata": {
+        "id": "F432oSPn2HSv"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "lib_path, model_path, chat_config_path = mlc_llm.build_model(build_args)"
+      ],
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "kMepfzs8O48L",
+        "outputId": "d5ddedd1-c871-418d-8af1-d77bf6ae0508"
+      },
+      "execution_count": 8,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "Using path \"Llama-2-7b-chat-hf\" for model \"Llama-2-7b-chat-hf\"\n",
+            "WARNING: --db-path does not point to a valid database: []\n",
```

Comment:
I checked that the default value of `--db-path` is still `"log_db"` in BuildArgs https://github.com/mlc-ai/mlc-llm/blob/d4c3a17379e1e890db6650568d7b5a59650a17bd/mlc_llm/core.py#L60. So just a bit curious why it emits this warning.

Reply:
Ohh I think this is because we are doing the model building in `my_workspace` instead of inside the `mlc_llm` repo. The intent is to show that, by installing the `mlc_llm` as a package, they can build wherever they want.

Reply:
I see. Sounds good for now, as we will remove the database soon when dlight gets maturer.

Reply:
Makes sense. Thanks for the suggestions!